sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +765,73 @@
             return monoError(logger, ex);
         }
     }
+
+    /**
+     * Initiates a search and returns a {@link PhoneNumberSearch} usable by other functions
+     * This function returns a Long Running Operation poller that allows you to 
+     * wait indefinitely until the operation is complete.
+     * 
+     * @param options A {@link CreateSearchOptions} with the search options
+     * @param lroDuration The time our long running operation will keep on polling 
+     * until it gets a result from the server
+     * @return A {@link PollerFlux} object with the search result
+     */
+    public PollerFlux<PhoneNumberSearch, PhoneNumberSearch> beginCreateSearch(
+        CreateSearchOptions options, Duration lroDuration) {
+        final Duration interval = lroDuration != null ? lroDuration : DEFAULT_POLL_INTERVAL;
+        return new PollerFlux<PhoneNumberSearch, PhoneNumberSearch>(interval,
+            createSearchActivationOperation(options),
+            createSearchPollOperation(),
+            (activationResponse, pollingContext) -> Mono.error(new RuntimeException("Cancellation is not supported")),
+            createSearchFetchResultOperation());
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PhoneNumberSearch>> createSearchActivationOperation(CreateSearchOptions options) {
+        return (pollingContext) -> {
+            Mono<PhoneNumberSearch> response = createSearch(options).flatMap(createSearchResponse ->
+            {
+                String searchId = createSearchResponse.getSearchId();
+                PhoneNumberSearch phoneNumberSearch = new PhoneNumberSearch().setSearchId(searchId);
```

Comment:
How is this going to work.. if you just assign the searchId to a new PhoneNumberSearch object? Shouldn't you do something - getSearchById(createSearchResponse.getSearchId())?